apps/frontend/src/app/Database/DataManagers/ArtifactData.ts

```diff
@@ -128,15 +112,22 @@ export class ArtifactDataManager extends DataManager<string, "artifacts", ICache
           }
           isUpgrade ? result.artifacts.upgraded.push(art) : result.artifacts.unchanged.push(art)
           idsToRemove.delete(match.id)
-          if (importId) {
-            delete this.data[match.id]
-            this.removeStorageEntry(match.id)
-          } else importId = match.id
+          if (importId) this.remove(match.id, false)// Do not notify, since this is a "replacement"
```

Comment:
Why are we removing?

Reply:
in this logic, we detected a duplicate/upgrade at `match.id` which will be "replaced" when the imported item is written to `importId` later. 

This is effective a "swap", where the old item(`match.id`) is deleted, and the new item(`importId`) is set.

Reply:
```suggestion
          // Detected a match. We are going to move the item at match.id to importId later, so remove it now to avoid a duplicate
          if (importId) this.remove(match.id, false)// Do not notify, since this is a "replacement"
```